include/ensmallen_bits/sgdr/cyclical_decay.hpp

```diff
@@ -96,7 +87,18 @@ class CyclicalDecay
       // Update the time for the next restart.
       nextRestart += epochRestart;
     }
+    // Time to adjust the step size.
+    if (epoch < nextRestart)
+    {
+      // n_t = n_min^i + 0.5(n_max^i - n_min^i)(1 + cos(T_cur/T_i * pi)).
+      stepSize = stepSizeMin + 0.5 * constStepSize *
+                 (1 + cos((( double ) batchRestart / epochRestart)
+                 * arma::datum::pi));
+      // Keep track of the number of batches since the last restart.
+      batchRestart++;
+    }
```

Comment:
This is a significant change to the functionality of `CyclicalDecay`.  If we are fixing a bug that is okay, but can you provide some justification please on what was wrong before, and what fixes it?  (justification in this comment is fine)

Also, the style is off here.  It should be `cos(((double) batchRestart / epochRestart)`.

Reply:
![image](https://user-images.githubusercontent.com/30979819/53369505-5dd77600-3971-11e9-9fcc-114d0b21e63e.png)
This is the update strategy as proposed in the paper .
1) The stepSizeMin parameter was added because , while in the original implmentation we assumed it to be 0 , the implmentation didnt really perform well as the learning rate slowed down too much near the ends of each epochBatch.
2) The step size change time , `epoch >= epochRestart` was changed because this is incorrect , we change every time before nextRestart , orginally this method changed it  for values of multiples of mult factor times eoochRestart , which isnt what the paper intended , we should do it until we reach next Restart .
From example of T0 = 10 and mult factor = 2, we changed for values less than 10, 20 ,40 ,80 and so on and not 10 ,30 ,70 etc.
The changes in the tests reflect this .

Reply:
Thanks for the clarification.  I understand now and agree that you've fixed a bug. :+1: